cpu/stm32f1/periph/gpio.c

```diff
@@ -90,6 +90,12 @@ int gpio_init(gpio_t pin, gpio_mode_t mode)
     port->CR[pin_num >> 3] &= ~(0xf << ((pin_num & 0x7) * 4));
     port->CR[pin_num >> 3] |=  ((mode & MODE_MASK) << ((pin_num & 0x7) * 4));
 
+    /* set ODR */
+    if (mode & 0x10)
```

Comment:
why not use `GPIO_IN_PU` then, instead of this magic number?

Reply:
or alternatively `GPIO_PULLUP`

Reply:
`GPIO_PULLUP` is something different. It is not even `0x10`, but `9`.

`GPIO_IN_PU` is `0x18` and thus also does not work.

I could define `MODE_PULLUP` with this value, next to `MODE_MASK`...

Reply:
okay forget about `GPIO_PULLUP`, but you wrote:

> The GPIO mode GPIO_IN_PU requires a bit to be set in a ODR register

so why not compare `mode` against that, it's the only mode that requires the ODR set. Hence I don't understand why

> GPIO_IN_PU is 0x18 and thus also does not work.

 IMHO there is no reason to use `0x10` here.

Reply:
done